qiskit/wrapper/_wrapper.py

```diff
@@ -21,35 +27,159 @@
 _DEFAULT_PROVIDER = DefaultQISKitProvider()
 
 
-def register(token, url='https://quantumexperience.ng.bluemix.net/api',
+def register(token=None, url='https://quantumexperience.ng.bluemix.net/api',
```

Comment:
Can we refactor a bit this function as well, aiming for simplifying the original use case of the function (ie. the current functionality) and treat the case of a missing token in a more modularized form? Similar to:

```
def register(token=None, ...):
    if not token:
        token, url, hub, ... = _load_token(...)
    provider = provider(token, ...)

def _load_token(....):
    # try _load_token_from_qconfig(), _load_token_from_envs(), _load_token_from_qiskitrc()
```

The idea is again to try to better encapsulate the functionality and the complexity (perhaps even moving the token-finding-in-all-places functionality as a whole to the `wrapper.configrc` module, renaming in a more generic form).

Reply:
Yes, it can be cleaned up.  I started adding functionality a bit hastily at the end.